backend/src/knowledge_table_api/services/vector_index/qdrant.py

```diff
@@ -0,0 +1,250 @@
+    path: Optional[str] = None
+
+
+class QdrantIndex(VectorIndex):
+    def __init__(self):
+        settings = Settings()
+        self.collection_name = settings.index_name
+        self.dimensions = settings.dimensions
+        self.client = QdrantClient(
+            **QdrantConfig().model_dump(exclude_none=True)
+        )
+        if not self.client.collection_exists(self.collection_name):
+            self.client.create_collection(
+                collection_name=self.collection_name,
+                vectors_config=models.VectorParams(
+                    size=self.dimensions, distance=models.Distance.COSINE
+                ),
+            )
+
+    async def upsert_vectors(
+        self, document_id: str, chunks: List[Document], llm_service: LLMService
+    ) -> Dict[str, str]:
+        entries = self.prepare_chunks(document_id, chunks, llm_service)
+        logger.info(f"Upserting {len(entries)} chunks")
+        points = [
+            models.PointStruct(
+                id=entry.pop("id"), vector=entry.pop("vector"), payload=entry
+            )
+            for entry in entries
+        ]
+        self.client.upsert(self.collection_name, points=points, wait=True)
+        return {"message": f"Successfully upserted {len(entries)} chunks."}
+
+    async def vector_search(
+        self, queries: List[str], document_id: str, llm_service: LLMService
+    ) -> dict[str, Any]:
+        logger.info(f"Retrieving vectors for {len(queries)} queries.")
+
+        embeddings = llm_service.get_embeddings()
+
+        final_chunks: List[Dict[str, Any]] = []
+
+        for query in queries:
+            logger.info("Generating embedding.")
+            embedded_query = np.array(embeddings.embed_query(query)).tolist()
+            logger.info("Searching...")
+
+            query_response = self.client.query_points(
+                self.collection_name,
+                query=embedded_query,
+                limit=40,
+                with_payload=True,
+                query_filter=models.Filter(
+                    must=[
+                        models.FieldCondition(
+                            key="document_id",
+                            match=models.MatchValue(value=document_id),
+                        )
+                    ]
+                ),
+            ).points
+
+            final_chunks.extend([point.payload for point in query_response])
+
+        seen_chunks, formatted_output = set(), []
+
+        for chunk in final_chunks:
+            if chunk["chunk_number"] not in seen_chunks:
+                seen_chunks.add(chunk["chunk_number"])
+                formatted_output.append(
+                    {"content": chunk["text"], "page": chunk["page_number"]}
+                )
+
+        logger.info(f"Retrieved {len(formatted_output)} unique chunks.")
+        return VectorResponse(
+            message="Query processed successfully.",
+            chunks=[Chunk(**chunk) for chunk in formatted_output],
+        )
```

Comment:
It looks like the `VectorResponse` class is used as the return type, but the return type annotation is `dict[str, Any]`. To maintain consistency and avoid potential type errors, consider changing the return statement to match the `dict[str, Any]` annotation. This will ensure the code accurately reflects the intended return type and aligns with parent class type annotations.